File: Solutions/1578. Minimum Deletion Cost to Avoid Repeating Letters/1578. Minimum Deletion Cost to Avoid Repeating Letters.cpp
class Solution {
public:
    int minCost(string s, vector<int>& cost) {
        int maxcost=0,res=0;
        for(int i=0;i<s.size();++i){
            if(i>0&&s[i]!=s[i-1])
                maxcost=0;
            res+=min(cost[i],maxcost);
            maxcost=max(cost[i],maxcost);
        }
        return res;
    }
};

File: Solutions/1578. Minimum Deletion Cost to Avoid Repeating Letters/1578. Minimum Deletion Cost to Avoid Repeating Letters.py
class Solution:
    def minCost(self, s: str, cost: List[int]) -> int:
        maxcost,res=0,0
        for i in range(len(s)):
            if i>0 and s[i]!=s[i-1]:
                maxcost=0
            res+=min(maxcost,cost[i])
            maxcost=max(maxcost,cost[i])
        return res